packages/data-table-server/src/dataTableService/utils/dataTableParamsToYupSchema.ts

```diff
@@ -15,7 +15,7 @@ const attachQualifiersToSchema = (
     qualifiedSchema = qualifiedSchema.oneOf(oneOf);
   }
   if (defaultValue) {
-    qualifiedSchema = qualifiedSchema.default(defaultValue);
+    qualifiedSchema = qualifiedSchema.default(schema.cast(defaultValue)); // This aims to cast date defaultValue from string type to date type
   } else if (required) {
     qualifiedSchema = qualifiedSchema.required();
```

Comment:
Date type defaultValue stored in data table config as string. It needs to be cast to Date before get used.